### PR TITLE
Cody: Add addition constrains to ripgrep

### PR DESCRIPTION
**Pull request by abeatrix**
RE: https://github.com/sourcegraph/sourcegraph/issues/51631

Add additional limits for the Ripgrep process to minimize the chance of freezing up users' machines:
- Includes all files except the ones that start with `.` or do not have a file extension, eg. `BUILD`, `LICENSE ` etc
- ~~Limits the search to 250 files~~
- Limits to use 1 thread
- Add timeout period to 30sec
- Skip files larger than 1M (based on [search.largeFiles](https://docs.sourcegraph.com/code_search/explanations/search_details#max-file-size))
- Remove `'-Tmarkdown', '-Tyaml', '-Tjson'` since those are covered by `-g *.*`

Also added logging for search performance time.

We can also look into adding `--max-depth`, `--max-count` if what we have implemented here does not help.

## Test plan

In my local instance, the returned context looks fine, but I need to figure out how to test if this helps with the unresponsiveness issue. Any advice is greatly appreciated 🙇  

image

### CODE REVIEW

**Comment by sourcegraph-buildkite** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.03 kb) | 0.00% (+0.03 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/bee/cody-rg/compare-report.html;tab=live_object) for a full comparison between the commits ce72e69 and 034db6e or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/cody/src/keyword-context/local-keyword-context-fetcher.ts [hunk 1]
```diff
@@ -8,7 +8,32 @@ import winkUtils from 'wink-nlp-utils'
 import { Editor } from '@sourcegraph/cody-shared/src/editor'
 import { KeywordContextFetcher, KeywordContextFetcherResult } from '@sourcegraph/cody-shared/src/keyword-context'
 
-const fileExtRipgrepParams = ['-Tmarkdown', '-Tyaml', '-Tjson', '-g', '!*.lock', '-g', '!*.snap']
+/**
+ * Exclude files without extensions and directories that starts with '.'
+ * Limits the search to 250 files
```
Comment (thenamankumar) [inline client/cody/src/keyword-context/local-keyword-context-fetcher.ts:13]: `--max-count` isn't max number of files. It is "Limit the number of matching lines per file searched to NUM."

Check `rg --help`

File: client/cody/src/keyword-context/local-keyword-context-fetcher.ts [hunk 2]
```diff
@@ -208,6 +233,7 @@ export class LocalKeywordContextFetcher implements KeywordContextFetcher {
                             {
                                 cwd: rootPath,
                                 maxBuffer: 1024 * 1024 * 1024,
+                                timeout: 1000 * 60, // timeout in one minute
```
Comment (thenamankumar) [inline client/cody/src/keyword-context/local-keyword-context-fetcher.ts:236]: I would go for 30s limit.

**Review by thenamankumar — APPROVED** [review]

File: client/cody/src/keyword-context/local-keyword-context-fetcher.ts (hunk 1 above)
Comment (abeatrix) [reply client/cody/src/keyword-context/local-keyword-context-fetcher.ts:13]: Ahh in my rg -h it prints so I assummed it means by files `-m, --max-count <NUM>                        Limit the number of matches.` but what you said makes sense. Thanks for catching that!

File: client/cody/src/keyword-context/local-keyword-context-fetcher.ts [hunk 3]
```diff
@@ -8,7 +8,29 @@ import winkUtils from 'wink-nlp-utils'
...
+    '-g',
+    '!.*',
+    '-g',
+    '!*.lock',
+    '-g',
+    '!*.snap',
+    '--threads',
+    '1',
+    '--max-filesize',
+    '1M',
```
Comment (camdencheek) [inline client/cody/src/keyword-context/local-keyword-context-fetcher.ts:32]: Nice, this should help a lot!

**Review by camdencheek — APPROVED** [review]

File: client/cody/src/keyword-context/local-keyword-context-fetcher.ts [hunk 4]
```diff
@@ -208,6 +231,7 @@ export class LocalKeywordContextFetcher implements KeywordContextFetcher {
                             {
                                 cwd: rootPath,
                                 maxBuffer: 1024 * 1024 * 1024,
+                                timeout: 1000 * 30, // timeout in one minute
```
Comment (mrnugget) [inline client/cody/src/keyword-context/local-keyword-context-fetcher.ts:234]: Comment is outdated

File: client/cody/src/keyword-context/local-keyword-context-fetcher.ts [hunk 5]
```diff
@@ -8,7 +8,29 @@ import winkUtils from 'wink-nlp-utils'
...
+/**
+ * Exclude files without extensions and directories that starts with '.'
+ * Limits to use 1 thread
+ * Exclude files larger than 1MB (based on search.largeFiles)
+ * Note: Ripgrep excludes binary files and respects .gitignore by default
+ */
+const fileExtRipgrepParams = [
+    '-Tmarkdown',
+    '-Tyaml',
+    '-Tjson',
```
[the remark this answers is not in the document]
Comment (abeatrix) [reply client/cody/src/keyword-context/local-keyword-context-fetcher.ts]: @camdencheek @mrnugget just removed these lines (in my latest commit) to make it less confusing but they are still included by `*'-g' '*.*'`.  

Confirmed by running a keyword search using this build, you can see the markdown files are included:
image

**Pull request merged**